src/utils.js

```diff
@@ -1412,6 +1413,26 @@ function syntaxErrorFactory(error) {
   return obj;
 }
 
+async function writeModulesMap(getJSON, resourcePath, exports) {
+  const json = exports.reduce((acc, { name, value }) => {
+    return { ...acc, [name]: value };
+  }, {});
+
+  const outputPath = path.resolve(
+    path.dirname(resourcePath),
+    `${path.basename(resourcePath)}.json`
+  );
+
+  if (getJSON === true) {
+    // If true, output a JSON CSS modules mapping file in the same directory as the resource
+    await fsp.writeFile(outputPath, JSON.stringify(json));
```

Comment:
Let's avoid it, we can't cover all possible cases, so `function` is better solution

Reply:
To clarify, do you mean only allowing the a function value for `getJSON` and removing the `true` option for it?  If so, I can definitely do that!